File: hr_ge_parser/frontend/src/types/job.ts
import type { PaginationMeta } from './common';
import type { Company } from './company';

// ============================================================
// BASE TYPES
// ============================================================

export interface Job {
  id: number;
  external_id: number;
  title: string;
  title_en: string | null;
  description: string | null;
  company: Company | null;
  company_id: number | null;
  slug: string | null;
  publish_date: string | null;
  deadline_date: string | null;
  renewal_date: string | null;
  salary_from: number | null;
  salary_to: number | null;
  salary_currency: string;
  show_salary: boolean;
  is_with_bonus: boolean;
  is_work_from_home: boolean;
  is_suitable_for_student: boolean;
  employment_type: string | null;
  work_schedule: string | null;
  contact_email: string | null;
  contact_phone: string | null;
  contact_name: string | null;
  hide_contact_person: boolean;
  is_expired: boolean;
  is_priority: boolean;
  application_method: number | null;
  languages: string[] | null;
  addresses: string[] | null;
  benefits: string[] | null;
  driving_licenses: string[] | null;
  created_at: string | null;
  updated_at: string | null;
  last_scraped_at: string | null;
}

// ============================================================
// API TYPES
// ============================================================

export interface JobFilters {
  page?: number;
  per_page?: number;
  search?: string;
  company_id?: number;
  is_expired?: boolean;
  is_work_from_home?: boolean;
  is_suitable_for_student?: boolean;
  salary_min?: number;
  salary_max?: number;
  location?: string;
  sort_by?: 'publish_date' | 'deadline_date' | 'salary_from' | 'created_at';
  sort_order?: 'asc' | 'desc';
}

export interface JobsResponse {
  data: Job[];
  meta: PaginationMeta;
}

// ============================================================
// UI TYPES
// ============================================================

export interface JobTableColumn {
  key: keyof Job | string;
  label: string;
  sortable?: boolean;
  width?: string;
}

File: hr_ge_parser/app/schemas/job.py
from datetime import datetime
from typing import Optional, List, Any
from pydantic import BaseModel, ConfigDict


class JobBase(BaseModel):
    title: str
    title_en: Optional[str] = None
    description: Optional[str] = None
    slug: Optional[str] = None
    publish_date: Optional[datetime] = None
    deadline_date: Optional[datetime] = None
    salary_from: Optional[int] = None
    salary_to: Optional[int] = None
    salary_currency: Optional[str] = "GEL"
    is_work_from_home: Optional[bool] = False
    is_suitable_for_student: Optional[bool] = False
    employment_type: Optional[str] = None
    work_schedule: Optional[str] = None
    is_expired: Optional[bool] = False


class JobCreate(JobBase):
    external_id: int
    company_id: Optional[int] = None


class CompanyInJob(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    external_id: int
    name: str
    name_en: Optional[str] = None
    logo_url: Optional[str] = None
    is_anonymous: Optional[bool] = False


class JobResponse(JobBase):
    model_config = ConfigDict(from_attributes=True)

    id: int
    external_id: int
    company: Optional[CompanyInJob] = None
    addresses: Optional[List[str]] = None
    languages: Optional[List[str]] = None
    created_at: Optional[datetime] = None


class JobDetail(JobResponse):
    model_config = ConfigDict(from_attributes=True)

    contact_email: Optional[str] = None
    contact_phone: Optional[str] = None
    contact_name: Optional[str] = None
    hide_contact_person: Optional[bool] = False
    is_priority: Optional[bool] = False
    is_with_bonus: Optional[bool] = False
    show_salary: Optional[bool] = True
    application_method: Optional[int] = None
    benefits: Optional[List[Any]] = None
    driving_licenses: Optional[List[Any]] = None
    renewal_date: Optional[datetime] = None
    last_scraped_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None


class PaginationMeta(BaseModel):
    total: int
    page: int
    per_page: int
    total_pages: int


class JobListResponse(BaseModel):
    data: List[JobResponse]
    meta: PaginationMeta
